### PR TITLE
Support Nested Components in CUE localizations

### PR DESCRIPTION
**Pull request by phoban01**
## Description

This PR adds support for aggregated components when performing complex Localization mappings.

When executing a complex mapping the `component` object now contains the following fields:

```cue
component: {
    version: string
    resources: [{}] // list of resources
    sources: [{}] // list of sources
    references: [{
        name: string // the name of the reference
        version: string // the version of the reference
        componentName: string // the name of the referenced component
        component: string // the full component descriptor for the referenced component
    }]
}
```

### CODE REVIEW

File: controllers/mutation_reconcile_looper_test.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
+package controllers
```
Comment (Skarlso) [inline controllers/mutation_reconcile_looper_test.go:1]: Nice!

File: main.go [hunk 2]
```diff
@@ -72,6 +72,10 @@ func main() {
 		os.Exit(1)
 	}
 
+	if v, found := os.LookupEnv("OCI_REGISTRY_LOCALHOST"); found {
```
Comment (Skarlso) [inline main.go:75]: What is the purpose of this? 🤔  Local testing? Or was this just left in?

**Review by Skarlso — APPROVED** [review]
LGTM, just one question. :) Well done! Nice job! :)

File: main.go (hunk 2 above)
Comment (phoban01) [reply main.go:75]: It's useful for local testing.

**Pull request merged**